roc/address.go

```diff
@@ -48,44 +49,46 @@ func NewAddress(family Family, ip string, port int) (*Address, error) {
 	if errCode < 0 {
 		return nil, ErrInvalidArguments
 	}
-	return nil, ErrInvalidApi
+
+	panic(fmt.Sprintf(
+		"unexpected return code %d from roc_address_init()", errCode))
 }
 
 // Family returns address family.
 //
 // If AfAuto was used to construct address, the actually selected family, i.e.
 // either AfIPv4 or AfIPv6, is reported.
-func (a *Address) Family() (Family, error) {
+func (a *Address) Family() Family {
 	f := C.roc_address_family(a.raw)
 	family := (Family)(f)
 	if family == afInvalid {
-		return family, ErrInvalidArguments
+		panic("unexpected failure in roc_address_family()")
```

Comment:
I don't think it's a good idea to panic here, it  will stop the work of the whole app afaik
the right thing to do here as a library is to report and error and move on
I do agree that in line 53 we should panic, because this situation when API return an unknown code is not possible and signalises that we don't follow the contract

Reply:
Well if we can panic on line 53, I think we can also panic here. roc_address_family() can fail (and return AF_INVALID) only if passed `address` is NULL or was not successfully initialized.

In the C API, the API itself allows the user to make such calls: the user can just pass NULL or uninitialized memory. As the API allows to do it, we handle it, and report error instead of blindly crashing.

However in the Go API, such case is not possible: NewAddress() will either fail or return non-null and fully initialized address object. So if the user has an object on which Family() can be called, Family() is guaranteed to succeed.

BTW I made `afInvalid` unexported for the same reason: it should never be visible by the user of Go API.

So if roc_address_family() fails here, it means that there is a bug either in C or Go part, just like on line 53.

So we should decide whether to panic on such "impossible" bugs detected in Go and C API. If you think the panic on line 53 is okay, then I guess this one is okay too.

Reply:
this doesn't feel right to me, but you have a point, address should be fully initialized
I think we can keep it like this and see what's going to come out of it

Reply:
OK, so for now we'll use the following approach: if we detect a violation of the C API contract, or violation of the internal invariant in Go bindings, we panic.